sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala

```diff
@@ -88,7 +98,7 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
   /**
    * @return [[SQLMetric]] for the `name`.
    */
-  def longMetric(name: String): SQLMetric = metrics(name)
+  def longMetric(name: String): SQLMetric = allMetrics(name)
```

Comment:
this looks expensive to me. Can we ask the caller side to get the specific metrics from `metrics` or `driverMetrics`?

Reply:
No problem, that's a clearer fix which only needs several lines code.